### ISSUE TITLE
RFC: add support for computing the cumulative product to the standard

### ISSUE THREAD

**Issue by steff456** [data-apis/array-api#598]
This RFC requests to include a new API in the array API specification for the purpose of computing the cumulative product.

## Overview

Based on array comparison [data](https://github.com/data-apis/array-api-comparison/blob/7ecfa9867ec43a28e914b35868c041443d31d772/data/common_apis.csv#L42), the API is available in the majority of the libraries in the PyData ecosystem.

## Prior art

- NumPy: <https://numpy.org/doc/stable/reference/generated/numpy.cumprod.html>
- PyTorch: https://pytorch.org/docs/stable/generated/torch.cumprod.html
- MXNet: <https://mxnet.apache.org/versions/master/api/python/docs/api/np/generated/mxnet.np.cumprod.html>
- TensorFlow: <https://www.tensorflow.org/api_docs/python/tf/math/cumprod>

## Proposal:

```python
def cumprod(x: array, /, *, axis: Optional[int] = None, dtype: Optional[dtype] = None) -> array
```

- `dtype` kwarg is for consistency with `sum` et al

cc @oleksandr-pavlyk

**Comment by kgryte** [comment]
@steff456 for the `axis` kwarg, this should be limited to just an `int` and not a `Tuple`. Based on the various docs, libraries seem to only support an `int`.

**Comment by kgryte** [comment]
One thing to note is that PyTorch, e.g., requires an axis argument and does not support NumPy's default behavior of computing the cumulative sum over the flattened array.
